### ISSUE TITLE
Fix the build

### ISSUE THREAD

**Issue by vtgruk** [widgetti/ipyaggrid#44]
The UI tests seem to be broken. Can we pin all the versions of the dependencies to get repeatable results?

**Comment by vthemelis** [comment]
Would it make sense to use some locking mechanism for this? Maybe use https://python-poetry.org ?

**Comment by maartenbreddels** [comment]
Hi,

I discussed this with @mariobuikhuizen today. Although we don't want to pin anything in the ipyagrid dependencies for users, we think it is good to do pinning/lockfile for CI.
On top of that we'd like to have a daily/nightly run of CI without a lockfile, and when succeeded, it should auto commit/update the lockfile.
This way CI does not randomly fail for a PR, and we can detect failures early on. Also, by committing the lockfile regularly, the 'diff' of the lockfile on failure will be small, and easy to analyse.
Does this summarize what we discussed @mariobuikhuizen ?

If we can set this up for ipyagrid, we'd like to do that for all the repo's in our organization.
I'm not sure if poetry makes sense, or a regular `pip freeze > requirements_ci.txt` is sufficient.

Regards,

Maarten
